kernel/src/table_changes/log_replay/tests.rs

```diff
@@ -0,0 +1,514 @@
+        .unwrap()
+        .into_iter();
+
+    let scanner = LogReplayScanner::prepare_table_changes(
+        commits.next().unwrap(),
+        engine.as_ref(),
+        &get_schema().into(),
+    )
+    .unwrap();
+    assert!(!scanner.has_cdc_action);
+    assert_eq!(scanner.remove_dvs, HashMap::new());
+
+    assert_eq!(
+        result_to_sv(scanner.into_scan_batches(engine, None).unwrap()),
+        &[true, true]
+    );
+}
+
+#[tokio::test]
+async fn filter_data_change() {
+    let engine = Arc::new(SyncEngine::new());
+    let mut mock_table = LocalMockTable::new();
+    mock_table
+        .commit([
+            Remove {
+                path: "fake_path_1".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_2".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_3".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_4".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+            // Add action that has same path as remove
+            Add {
+                path: "fake_path_1".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+            // Add action with unique path
+            Add {
+                path: "fake_path_5".into(),
+                data_change: false,
+                ..Default::default()
+            }
+            .into(),
+        ])
+        .await;
+
+    let mut commits = get_segment(engine.as_ref(), mock_table.table_root(), 0, None)
+        .unwrap()
+        .into_iter();
+
+    let scanner = LogReplayScanner::prepare_table_changes(
+        commits.next().unwrap(),
+        engine.as_ref(),
+        &get_schema().into(),
+    )
+    .unwrap();
+    assert!(!scanner.has_cdc_action);
+    assert_eq!(scanner.remove_dvs, HashMap::new());
+
+    assert_eq!(
+        result_to_sv(scanner.into_scan_batches(engine, None).unwrap()),
+        &[false; 6]
+    );
+}
+
+#[tokio::test]
+async fn cdc_selection() {
+    let engine = Arc::new(SyncEngine::new());
+    let mut mock_table = LocalMockTable::new();
+    mock_table
+        .commit([
+            Add {
+                path: "fake_path_1".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_2".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+            Cdc {
+                path: "fake_path_3".into(),
+                data_change: true,
```

Comment:
doesn't CDC always have `data_change: false`?
```suggestion
                data_change: false,
```